propagation.go

```diff
@@ -57,53 +59,46 @@ type Extractor interface {
 	Extract(carrier interface{}) (SpanContext, error)
 }
 
-type textMapPropagator struct {
-	tracer      *tracer
-	encodeValue func(string) string
-	decodeValue func(string) string
+type coder interface {
```

Comment:
per the comment below, if zipkin is moved into its own subpackage, I think you can restore this file completely to its current form, no reason to introduce changes unless necessary. Also, I do not believe zipkin propagator needs to use `url.QueryEscape(val)` because its values are already safe, whereas in Jaeger we used `:` as a separator which does get url-encoded.